File: SortingAlgos/SelectionSort.java
class SelectionSort {

    public static int[] sort(int[] arr) {
        return selectionSort(arr);
    }

    public static int[] selectionSort(int[] arr) {
        // time O(n^2)
        // un-stable
        // inplace
        
        for (int lastUnsortedIndex = arr.length - 1; lastUnsortedIndex > 0; lastUnsortedIndex--) {

            int largestIndex = 0;

            for (int j = 1; j <= lastUnsortedIndex; j++) {
                if (arr[j] > arr[largestIndex]) {
                    largestIndex = j;
                }
                swap(arr, largestIndex, lastUnsortedIndex);
            }
        }

        return arr;
    }

    private static void swap(int[] arr, int i, int j) {

        if (i == j) {
            return;
        }

        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

}

class testSelection {

    public static void main(String args[]) {

        int[] arr = { 20, 35, -11, 7, 55, 1, -22 };

        SelectionSort.sort(arr);

        for (int i : arr) {
            System.out.print(i + ", ");
        }

        System.out.println();

    }
}

File: SortingAlgos/SelectionSort.py
class SelectionSort(object):
    
    def sort(self, arr):
        return self.selectionSort(arr)

    def selectionSort(self, arr) -> list:

        lastUnsortedIndex = len(arr)-1

        while (lastUnsortedIndex >= 0):
            largest = 0
            for i in range(lastUnsortedIndex+1):
                if arr[i] > arr[largest]:
                    largest = i
            
            self.swap(arr,lastUnsortedIndex,largest)
            lastUnsortedIndex -= 1

        return arr
    

    def swap(self, arr, i, j):
        
        if i == j:
            return
            
        temp = arr[i]
        arr[i] = arr[j]
        arr[j] = temp



arr = [20, 35, -15, 7, 55, 1, -22]
selectionSort = SelectionSort()

print(selectionSort.sort(arr))